sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -246,6 +246,12 @@ def set_initial_status(self, pipeline_response: "PipelineResponseType") -> str:
 
         self._set_async_url_if_present(response)
 
+        # Check if we can extract status from initial response, if present
+        try:
```

Comment:
Should we do this in 

if response.status_code in {200, 201, 202, 204} and self._async_url:

?

Reply:
Fair feedback